pkg/metrics/metrics.jsx

```diff
@@ -292,6 +308,12 @@ class CurrentMetrics extends React.Component {
             this.temperature_channel.addEventListener("message", this.onTemperatureUpdate);
         }
 
+        // try privileged channel so system administrator can see per-service IO usage of all users
+        if (!cockpit.hidden && (this.privileged_channel === null)) {
+            this.privileged_channel = cockpit.channel({ superuser: "try", payload: "metrics1", source: "internal", interval: 3000, metrics: PRIVILEGED_METRICS });
```

Comment:
@martinpitt Since I need privileged access to be able to sample every process in /proc/<pid> should I open new channel with `"superuser": "try"` or should I reuse the existing metrics channel and add superuser to it?

I assume limiting what runs with escalated permissions is the safer call here and opening one extra channel won't add too much overhead.

Reply:
Is this question related to this particular `metrics1` channel? I suppose not, as you want to read files from /proc directly, just making sure we talk about the same thing.

Please do the same thing for reading /proc, with superuser=try. Then admins can scan all processes, and non-admins can at least scan their own. But please don't open hundreds of channels to read all /proc files directly, run a helper shell or python script that scrapes all the info and outputs a JSON stream (so that it keeps running).

Reply:
Yes, it's about this one channel. c-bridge is who reads from /proc and then sends data through this channel as a JSON stream.